src/libvirtApi/domain.js

```diff
@@ -452,18 +454,34 @@ export function domainDetachDisk({
             });
 }
 
-export function domainDetachHostDevice({ connectionName, vmName, live, dev }) {
-    const options = { err: "message" };
-    const source = getNodeDevSource(dev);
-    const args = ["virt-xml", "-c", `qemu:///${connectionName}`, vmName, "--remove-device", "--hostdev", source];
+export function domainDetachHostDevice({ connectionName, vmName, vmId, live, dev }) {
```

Comment:
It's redundant to pass `vmName` as the function already requires `vmId` anyway.  It also does not use `vmName` anywhere, so please drop it.